fix makefile and cmake logic for AARCH64

Haus1 opened the pull request:
Some ifndef slipped through in place of ifdefs and are throwing the compiler for a loop.

Use ifdef to match inclusive filters in the Makefile and have GGML_CPU_AARCH64 default to OFF. This prevents the compiler from becoming confused and optimizing for the wrong architecture.


https://github.com/ggerganov/llama.cpp/issues/11204
https://github.com/ggerganov/llama.cpp/issues/11082
https://github.com/ggerganov/llama.cpp/issues/11083
https://github.com/ggerganov/llama.cpp/issues/11079
https://github.com/ggerganov/llama.cpp/issues/11071
https://github.com/ggerganov/llama.cpp/issues/10933

slaren commented:
`GGML_CPU_AARCH64` enabled by default is not a bug. The name of the feature is misleading, but it also supports AVX2.

Haus1 commented:
Is this in addition to what's enabled with GGML_AVX2?

Part of the motivation behind this is the build failing for clang versions prior to 18, specifically the ROCM compiler. These changes fall short of fixing it but everything I've found points to it being an issue with clang's handling of aarch64 and getting stuck in an infinite loop.

Haus1 committed:
fix makefile logic for architecture specific flags

Some ifndef slipped through in place of ifdefs and are throwing the compiler for a loop.
Use ifdef to match inclusive filters in the Makefile.

Haus1 commented:
Updated to include a few more for arm that somehow got missed and removed the changes to GGML_CPU_AARCH64.

The indenting and formatting of the Makefile can be a bit confusing/inconsistent but that's beyond the scope here
